pkg/apis/maistra/v1/servicemeshcontrolplane_types.go

```diff
@@ -44,21 +44,28 @@ type ControlPlaneStatus struct {
 	StatusType `json:",inline"`
 	// ComponentStatus represents the current status of the components
 	ComponentStatus []*ComponentStatus `json:"components"`
+
+	//LastAppliedConfiguration lists the last appllied ServiceMeshControlPlane
+	LastAppliedConfiguration ControlPlaneSpec `json:"lastAppliedConfiguration"`
 }
 
 // HelmValuesType is typedef for Helm .Values
 type HelmValuesType map[string]interface{}
 
 // ControlPlaneSpec represents the configuration for installing a control plane
 type ControlPlaneSpec struct {
+	//Template selects the template to use for default values
+	Template string `json:"template,omitempty"`
```

Comment:
Would it make sense to have this be a list?  For example, a user specifies something like:

```yaml
spec:
  template:
  - full
  - auth
```

Reply:
Think we discussed this with Kevin before and decided not to for now. This would also bring up questions like in what order do things overlay each other. Instead we have a hierarchical inheritance.

Reply:
complication here is that it would require an api change, so better to decide now, as we'll have to live with it for a while.

as for the logic, why not apply them in order, full then auth, given the example above.  nothing complicated about that.

Reply:
adding some color, this allows for composition, which avoids an expansion of templates (full-auth, minimal-auth, etc.), "prefer composition over inheritance."